src/psyclone/psyir/nodes/array_mixin.py

```diff
@@ -143,59 +205,49 @@ def is_lower_bound(self, index):
         self._validate_index(index)
 
         array_dimension = self.indices[index]
-        if not isinstance(array_dimension, Range):
-            return False
 
-        lower = array_dimension.start
-
-        if isinstance(lower, Literal):
+        if not isinstance(array_dimension, Range):
+            # Check for a single element array
             try:
                 symbol = self.scope.symbol_table.lookup(self.name)
```

Comment:
I appreciate this line hasn't been modified here but I don't understand what's going on. First, why can't we just do `self.symbol` and second, what happens if this is a member of a structure?

Reply:
This is looking for the case where the array dimension declaration is of size 1 so the whole of the array can be accessed without using a Range.

Reply:
What seems to be happening is that the exception happens when we have a member of a structure, so essentially the code captures it being a member of a structure and causes it to give up. Obviously this is not the right way to do this. I just need to either give up on structures for the moment or find out how to get access to the size of the array within the structure.

Reply:
I've re-written as I agree it was confusing. Structures are OK when we use array notation but not for an array access. I find getting the declaration information from an array within a structure rather difficult so have left that for the moment.